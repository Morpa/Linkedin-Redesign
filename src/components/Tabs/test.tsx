import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import TabsContent from '.'

const props = {
  firstTitle: 'Profile',
  secondTitle: 'Activity & Interests',
  thirdTitle: 'Articles',
  firstContent: 'Content 01',
  secondContent: 'Content 02',
  thirdContent: 'Content 03'
}

describe('<Tabs />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<TabsContent {...props} />)

    expect(screen.getByRole('tab', { name: /profile/i })).toBeInTheDocument()

    expect(
      screen.getByRole('tab', { name: /activity & interests/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('tab', { name: /articles/i })).toBeInTheDocument()

    expect(screen.getByRole('tab', { name: /profile/i })).toHaveClass(
      'react-tabs__tab react-tabs__tab--selected'
    )

    expect(screen.getByRole('tab', { name: /articles/i })).not.toHaveClass(
      'react-tabs__tab react-tabs__tab--selected'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
