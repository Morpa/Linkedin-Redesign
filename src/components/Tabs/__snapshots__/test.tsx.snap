// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render the heading 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
}

.c0 .react-tabs {
  -webkit-tap-highlight-color: transparent;
}

.c0 .react-tabs__tab-list {
  border-bottom: 0.1rem solid #CECECE;
  margin: 0 0 4.0rem;
  padding: 0;
}

.c0 .react-tabs__tab {
  width: 24rem;
  height: 4rem;
  display: inline-block;
  border: 0.1rem solid #CECECE;
  color: #181818;
  background: #FFFFFF;
  text-align: center;
  border-bottom: 0.1rem solid #CECECE;
  bottom: -1.2rem;
  position: relative;
  list-style: none;
  padding: 1.6rem 0;
  cursor: pointer;
  font-weight: 600;
  font-size: 1.6rem;
  text-transform: uppercase;
}

.c0 .react-tabs__tab--selected {
  width: 24rem;
  height: 5rem;
  bottom: 0;
  border: none;
  border-radius: 0.4rem 0.4rem 0 0;
  background: #0275B1;
  color: #FFFFFF;
}

.c0 .react-tabs__tab--disabled {
  cursor: default;
}

.c0 .react-tabs__tab:focus {
  box-shadow: 0 0 5px hsl(208,99%,50%);
  border-color: hsl(208,99%,50%);
  outline: none;
}

.c0 .react-tabs__tab:focus:after {
  content: '';
  position: absolute;
  height: 5px;
  left: -4px;
  right: -4px;
  bottom: -5px;
  background: #fff;
}

.c0 .react-tabs__tab-panel {
  display: none;
}

.c0 .react-tabs__tab-panel--selected {
  display: block;
}

<main
  class="c0"
>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        Profile
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        id="react-tabs-2"
        role="tab"
      >
        Activity & Interests
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        id="react-tabs-4"
        role="tab"
      >
        Articles
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      Content 01
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel"
      id="react-tabs-5"
      role="tabpanel"
    />
  </div>
</main>
`
